frame/src/routes/gallery.tsx

```diff
@@ -1,15 +1,19 @@
-import { Button, Frog } from 'frog'
-import { getCollection, getItem } from '../services/uniquery'
-import { kodaUrl } from '../utils'
 import { $purifyOne } from '@kodadot1/minipfs'
+import { Button, Frog } from 'frog'
 import { HonoEnv } from '../constants'
+import { getCollection, getItemByOffset } from '../services/uniquery'
+import { kodaUrl } from '../utils'
 
 export const app = new Frog<HonoEnv>({})
 
 app.frame('/:chain/:id', async (c) => {
   const { chain, id } = c.req.param()
-  const collection = await getCollection(chain, id)
-  console.log({ collection })
+  const [collection, firstItemInCollection] = await Promise.all([
+    getCollection(chain, id),
+    getItemByOffset(chain, id, 0),
+  ]);
+  const firstItemId = firstItemInCollection.id.split("-")[1];
+
```

Comment:
I would keep as it was and pass 0 anyway

```suggestion
  const collection = await getCollection(chain, id)
```